app/services/publish/accredited_provider_updater.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Publish
+  class AccreditedProviderUpdater
+    attr_reader :provider, :new_accredited_provider, :recruitment_cycle
+
+    def initialize(provider_code:, recruitment_cycle_year:, new_accredited_provider_code:)
+      @recruitment_cycle = RecruitmentCycle.find_by!(year: recruitment_cycle_year)
+      @provider = Provider.find_by!(provider_code:, recruitment_cycle: @recruitment_cycle)
+      @new_accredited_provider = Provider.find_by!(provider_code: new_accredited_provider_code, recruitment_cycle: @recruitment_cycle)
+    end
+
+    def update_provider_and_courses
+      update_provider && update_courses
+    end
+
+    def update_provider
+      provider.update_columns(accrediting_provider: 'not_an_accredited_provider',
+                              accrediting_provider_enrichments: new_accrediting_provider_enrichments)
+    end
+
+    def update_courses
+      courses = provider.courses
+      courses.update_all(accredited_provider_code: new_accredited_provider.provider_code)
```

Comment:
If a provider can be accredited by multiple providers, this code could cause unintended behavior by replacing the accreditation for all courses, even those that might need to retain their original accredited provider.

Example: Provider A has courses that are accredited by Provider B and Provider C.
After running this code to set the accredited_provider_code to Provider D, all courses will be accredited by Provider D, which may overwrite the correct accreditation for the courses that were accredited by Provider B or Provider C.

This is my first time reading through the accredited provider on the Publish side so I could be missing some parts.

Is it expected?

Reply:
This is a good point. Maybe we should only update the courses that are accredited by the old accredited provider, I'll update the PR 👌

Reply:
We would need to identify which accredited provider we were removing in order to specify which courses to update / not update.

For the immediate concern of updating specific providers to correct the data on behalf of the register team I think we can commit this service as is. 

In future, we could add a wizard service that prompted the support user to choose which courses should and should not have their accredited provider updated. This goes for the user permissions that exist on the providers, which also requires contextual selection.